cve_bin_tool/checkers/bzip2.py

```diff
@@ -12,8 +12,14 @@
 
 
 class Bzip2Checker(Checker):
-    # FIXME: fix contains pattern
-    CONTAINS_PATTERNS = []
+    CONTAINS_PATTERNS = [
+        r"bzip2recover ([0-9]+\.[0-9]+\.[0-9]+): extracts blocks from damaged .bz2 files.",
+        r"%s: BZ_MAX_HANDLED_BLOCKS in bzip2recover.c, and recompile.",
+        # r"in the bzip2-1.0.6 source distribution.", # present only .rpm
+    ]
     FILENAME_PATTERNS = [r"bzip2"]
-    VERSION_PATTERNS = [r"bzip2-([0-9]+\.[0-9]+\.[0-9]+)"]
+    VERSION_PATTERNS = [
+        r"bzip2-([0-9]+\.[0-9]+\.[0-9]+)",
+        r"bzip2recover ([0-9]+\.[0-9]+\.[0-9]+)",
+    ]
```

Comment:
I was going with the version pattern `r"bzip2(recover)?[ -]([0-9]+\.[0-9]+\.[0-9]+)"`, but this was resulting in failing of all tests. 

```
version_patterns = [re.compile('bzip2(recover)?[ -]([0-9]+\\.[0-9]+\\.[0-9]+)')]

    def regex_find(lines, version_patterns) -> str:
        """Search a set of lines to find a match for the given regex"""
        new_guess = ""

        for line in lines:
            for pattern in version_patterns:
                match = pattern.search(line)
                if match:
>                   new_guess2 = match.group(1).strip()
E                   AttributeError: 'NoneType' object has no attribute 'strip'

cve_bin_tool/util.py:86: AttributeError
```

Should we add a warning stating this?

Reply:
it seems like version is not detected in that pattern hence the `NoneType` in `group(1)`.
We may not need that pattern, since it doesn't work.

Reply:
@BreadGenie This is what I am receiving as in version strings for .rpm,

```
VERSION_PATTERNS = [
        bzip2recover 1.0.6: extracts blocks from damaged .bz2 files.,
        in the bzip2-1.0.6 source distribution.,
]
```
and this for .deb
```
VERSION_PATTERNS = [
        bzip2recover 1.0.8: extracts blocks from damaged .bz2 files.,
]
```
I think we could just add a warning of some sort like "do not use optional groups (which have ? (0 or 1 matches))" in the start of regex

Also, I should've put this error earlier :(

```
    def test_version_in_package(self, url, package_name, product, version):
        """Helper function to get a file (presumed to be a real copy
        of a library, probably from a Linux distribution) and run a
        scan on it.  Any test using this should likely be listed as a
        long test."""
        # get file
        tempfile = self.condensed_filepath(url, package_name)
        # new scanner for the new test.
        # self.scanner = VersionScanner(self.cve_scanner, should_extract=True)
        # run the tests
        list_products = set()
        list_versions = set()
        for scan_info in self.scanner.recursive_scan(tempfile):
            if scan_info:
                product_info, file_path = scan_info
                list_products.add(product_info.product)
                list_versions.add(product_info.version)

        # Make sure the product and version are in the results
        assert (
            product in list_products
        ), f"""{product} not found in {package_name}.
        The checker signature or url may be incorrect."""
>       assert version in list_versions
E       AssertionError: assert '1.0.8' in {'recover'}
```

Reply:
@BreadGenie This is what I am receiving as in version strings for .rpm,

```
VERSION_PATTERNS = [
        bzip2recover 1.0.6: extracts blocks from damaged .bz2 files.,
        in the bzip2-1.0.6 source distribution.,
]
```
and this for .deb
```
VERSION_PATTERNS = [
        bzip2recover 1.0.8: extracts blocks from damaged .bz2 files.,
]
```
I think we could just add a warning of some sort like "do not use optional groups (which have ? (0 or 1 matches))" in the start of regex

Also, I should've put this error earlier :(

```
    def test_version_in_package(self, url, package_name, product, version):
        """Helper function to get a file (presumed to be a real copy
        of a library, probably from a Linux distribution) and run a
        scan on it.  Any test using this should likely be listed as a
        long test."""
        # get file
        tempfile = self.condensed_filepath(url, package_name)
        # new scanner for the new test.
        # self.scanner = VersionScanner(self.cve_scanner, should_extract=True)
        # run the tests
        list_products = set()
        list_versions = set()
        for scan_info in self.scanner.recursive_scan(tempfile):
            if scan_info:
                product_info, file_path = scan_info
                list_products.add(product_info.product)
                list_versions.add(product_info.version)

        # Make sure the product and version are in the results
        assert (
            product in list_products
        ), f"""{product} not found in {package_name}.
        The checker signature or url may be incorrect."""
>       assert version in list_versions
E       AssertionError: assert '1.0.8' in {'recover'}
```

Reply:
So I think it's best to split that regex 2 regex, one without `recover` and one with `recover`

Reply:
Yeah, I did that, but it took me a while to figure out why it was happening. So, I think we could maybe add a warning message for this of some kind. Waiting for Mentors thoughts on this.